pkg/controllers/nodepool/status/readiness.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+)
+
+type Readiness struct {
+	cloudProvider cloudprovider.CloudProvider
+	kubeClient    client.Client
+}
+
+func (n Readiness) Reconcile(ctx context.Context, nodePool *v1beta1.NodePool) (reconcile.Result, error) {
+	nodePool.StatusConditions().SetFalse(v1beta1.ConditionTypeNodePoolReady, "NodePoolNotReady", "unable to resolve underlying nodeClass")
+	nodeClassList := &unstructured.UnstructuredList{}
```

Comment:
Should this just be a Get()? I think the NodePool can only reference a single NodeClass